elastic-buffer-sim/src/Bittide/Simulate.hs

```diff
@@ -119,7 +116,7 @@ elasticBuffer mode size clkRead clkWrite =
   goWrite ticks fillLevel = go ticks newFillLevel
    where
     newFillLevel
-      | fillLevel >= size = case mode of
+      | fillLevel == maxBound = case mode of
           Saturate -> fillLevel
           Error -> error "elasticBuffer: overflow"
```

Comment:
Why do we want `error` and not `errorX`?

Reply:
Because we want to kill the simulation when this happens. (This will change in a later PR, where elastic buffers have to deal with overflows in "boot mode".)